### PR TITLE
Fixup autotuning sample config

### PR DESCRIPTION
**Pull request by erikjohnston**
This was added in #12701

### CODE REVIEW

File: docs/sample_config.yaml [hunk 1]
```diff
@@ -784,22 +784,33 @@ caches:
   #
   #cache_entry_ttl: 30m
 
-  # This flag enables cache autotuning, and is further specified by the sub-options `max_cache_memory_usage`,
-  # `target_cache_memory_usage`, `min_cache_ttl`. These flags work in conjunction with each other to maintain
-  # a balance between cache memory usage and cache entry availability. You must be using jemalloc to utilize
-  # this option, and all three of the options must be specified for this feature to work.
+  # This flag enables cache autotuning, and is further specified by the
```
Comment (richvdh) [inline docs/sample_config.yaml:787]: to me, "flag" is a boolean. None of these things are flags?  Not a major problem though.

File: docs/sample_config.yaml [hunk 2]
```diff
@@ -784,22 +784,33 @@ caches:
...
-    # 'min_cache_ttl` sets a limit under which newer cache entries are not evicted and is only applied when
-    # caches are actively being evicted/`max_cache_memory_usage` has been exceeded. This is to protect hot caches
-    # from being emptied while Synapse is evicting due to memory.
+    # 'min_cache_ttl` sets a limit under which newer cache entries are not
+    # evicted and is only applied when caches are actively being
+    # evicted/`max_cache_memory_usage` has been exceeded. This is to
+    # protect hot caches from being emptied while Synapse is evicting due
+    # to memory.
+    #
     #min_cache_ttl: 5m
```
Comment (richvdh) [inline docs/sample_config.yaml:814]: what are the defaults for these things? (what happens if they are not set?) (also needs adding to the config manual I think)

File: docs/sample_config.yaml [hunk 3]
```diff
@@ -784,22 +784,33 @@ caches:
...
-    # This flag sets a ceiling on much memory the cache can use before caches begin to be continuously evicted.
-    # They will continue to be evicted until the memory usage drops below the `target_memory_usage`, set in
-    # the flag below, or until the `min_cache_ttl` is hit.
+    # This flag sets a ceiling on much memory the cache can use before
+    # caches begin to be continuously evicted. They will continue to be
+    # evicted until the memory usage drops below the
+    # `target_memory_usage`, set in the flag below, or until the
+    # `min_cache_ttl` is hit.
+    #
     #max_cache_memory_usage: 1024M
```
Comment (richvdh) [inline docs/sample_config.yaml:801]: please could you document, here and in the config manual, what the valid suffixes are, and what they mean? (is M 1000000 or 1048576 ?)

**Review by richvdh — CHANGES_REQUESTED** [review]

File: docs/sample_config.yaml (hunk 1 above)
Comment (richvdh) [reply docs/sample_config.yaml:787]: s/flag/setting/ would be better, I think

**Comment by H-Shay** [comment]
I can fix all this in the config manual, as it was on my agenda to rip out the config comments today anyway.

**Comment by H-Shay** [comment]
The issues have hopefully been addressed here: https://github.com/matrix-org/synapse/pull/12776

I would say that this should be closed as I am in the process of deleting all comments from the config file as we speak.

**Pull request closed without merging**